jcl/src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -376,6 +376,53 @@ private static final MethodType fromMethodDescriptorStringAppendArg(String metho
 		return methodType(returnType, types);
 	}
 	
+	/**
+	 * This helper calls MethodType.fromMethodDescriptorString(...) or 
+	 * MethodType.fromMethodDescriptorStringAppendArg(...) but throws 
+	 * NoClassDefFoundError instead of TypeNotPresentException during 
+	 * the VM resolve stage.
+	 *
+	 * @param methodDescriptor - the method descriptor string
+	 * @param loader - the ClassLoader to be used
+	 * @param appendArgumentType - an extra argument type
+	 *
+	 * @return a MethodType object representing the method descriptor string
+	 *
+	 * @throws IllegalArgumentException - if the string is not well-formed
+	 * @throws NoClassDefFoundError - if a named type cannot be found
+	 */
+	static final MethodType vmResolveFromMethodDescriptorString(String methodDescriptor, ClassLoader loader, Class<?> appendArgumentType) throws Throwable {
+		try {
+			if (null == appendArgumentType) {
+				return MethodType.fromMethodDescriptorString(methodDescriptor, loader);
+			}
+			return MethodType.fromMethodDescriptorStringAppendArg(methodDescriptor, loader, appendArgumentType);
+		} catch (TypeNotPresentException e) {
+			throw throwNoClassDefFoundError(e);
+		}
+	}
+
+	/**
```

Comment:
Can you fix the formatting for the javadoc on this and the above method?  The formatting is inconsistent between the two.

Reply:
fixed.